components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -327,7 +328,13 @@ public Application updateApplication(ApplicationUpdateRequest updateRequest, Str
             }
             appDTO.setRequestObjectSignatureValidationEnabled(updateRequest.isRequireSignedRequestObject());
             appDTO.setRequirePushedAuthorizationRequests(updateRequest.isRequirePushedAuthorizationRequests());
-            appDTO.setTlsClientCertificateBoundAccessTokens(updateRequest.isTlsClientCertificateBoundAccessTokens());
+            if (updateRequest.isTlsClientCertificateBoundAccessTokens()) {
+                if (Boolean.parseBoolean(IdentityUtil.getProperty(
+                        OAuthConstants.ENABLE_TLS_CERT_BOUND_ACCESS_TOKENS_VIA_BINDING_TYPE))) {
```

Comment:
instead of checking for the config here lets check if the CERTIFICATE_BASED_TOKEN_BINDER component is available and then set it.

Reply:
changed